fix: make match logic consider any match

DaJoTo opened the pull request:
This should fix #1865

DaJoTo commented on lib/runner/matchers/filename.js:
```diff
@@ -114,4 +116,4 @@ class FilenameMatcher {
   }
```
I had considered this
```javascript
match(fullPath) {
  return this.filterPattern.some(pattern => fullPath.includes(pattern) || minimatch(fullPath, pattern));
}
```
but the PR as it is felt more in keeping with the current style, however happy to change if you prefer

beatfactor replied:
Yes, this is significantly more elegant and also better performance.

beatfactor commented:
@DaJoTo do you think it's possible to add a test also? There are some exclude tests in https://github.com/nightwatchjs/nightwatch/blob/master/test/src/runner/testRunWithExclude.js.

DaJoTo commented:
Absolutely, I'll *try* and get it done before the weekend

DaJoTo commented:
@beatfactor I added two tests, perhaps one would suffice? Sorry for the delay. To be honest I've not used mocha before and have little experience with testing in javascript. You'll note I refactored the test file and am happy to change this back if it's the wrong thing to do with tests.
